monarchmoney/monarchmoney.py

```diff
@@ -2352,14 +2392,54 @@ async def set_budget_amount(
             graphql_query=query,
         )
 
+    async def upload_and_parse_balance_history(
+        self,
+        account_id: str,
+        csv_content,
+        timeout: int = 300,
+        delay: int = 10,
+    ) -> bool:
+        """
+        Uploads and parses the balance history, updating the account balance on monarch money
+
+        :param account_id: The account ID to apply the history to.
+        :param csv_content: CSV representation of the balance history. Headers are Date, Amount, and Account Name.
+        :param timeout: The number of seconds to wait before timing out
+        :param delay: The number of seconds to wait for each check on whether parsing is completed
+        """
+
+        session_key = await self.upload_account_balance_history(
+            account_id=account_id, csv_content=csv_content
+        )
+
+        response = await self.parse_upload_balance_history_session(
+            session_key=session_key
+        )
+
+        is_completed = (
+            response["parseBalanceHistory"]["uploadBalanceHistorySession"]["status"]
+            == "completed"
+        )
+
+        start = time.time()
+
+        while not is_completed and (time.time() <= (start + timeout)):
+            await asyncio.sleep(delay)
+
+            is_completed = (await self.get_upload_balance_history_session(session_key))[
+                "uploadBalanceHistorySession"
+            ]["status"]
+
+        return is_completed
+
     async def upload_account_balance_history(
         self, account_id: str, csv_content: str
-    ) -> None:
+    ) -> str:
         """
-        Uploads the account balance history csv for a given account.
+        Uploads the account balance history CSV for a specified account.
 
         :param account_id: The account ID to apply the history to.
-        :param csv_content: CSV representation of the balance history.
+        :param csv_content: CSV representation of the balance history. Headers: Date, Amount, and Account Name.
```

Comment:
nit: move this description into the body of the docblock, instead of here, where it's challenging to fit.

Reply:
done